### PR TITLE
Move service cert signer to operator v1 spec and status

### PR DESCRIPTION
**Pull request by enj**
Signed-off-by: Monis Khan <mkhan@redhat.com>

@deads2k @smarterclayton @openshift/sig-auth

### CODE REVIEW

File: servicecertsigner/v1alpha1/types.go [hunk 1]
```diff
@@ -14,13 +15,7 @@ import (
...
 	metav1.TypeMeta `json:",inline"`
 
-	// ServingInfo is the HTTP serving information for the controller's endpoints
-	ServingInfo configv1.HTTPServingInfo `json:"servingInfo,omitempty"`
-
-	// authentication allows configuration of authentication for the endpoints
-	Authentication DelegatedAuthentication `json:"authentication,omitempty"`
-	// authorization allows configuration of authentication for the endpoints
-	Authorization DelegatedAuthorization `json:"authorization,omitempty"`
+	operatorv1alpha1.GenericOperatorConfig `json:",inline"`
```
Comment (smarterclayton) [inline servicecertsigner/v1alpha1/types.go]: Is this a human facing config object or the internal operator driven config object?

[the remark this answers is not in the document]
Comment (enj) [reply servicecertsigner/v1alpha1/types.go]: Ah, same fields, but much clearer use case.

[the remark this answers is not in the document]
Comment (enj) [reply servicecertsigner/v1alpha1/types.go]: > I wonder what that embedded typemeta does. that could be exciting.

It appears to work...

Comment (enj) [reply servicecertsigner/v1alpha1/types.go]: Added godocs and moved to `GenericControllerConfig`

**Pull request merged**